plugins/woocommerce-blocks/assets/js/blocks/product-collection/constants.ts

```diff
@@ -56,7 +56,7 @@ export const DEFAULT_QUERY: ProductCollectionQuery = {
 	taxQuery: {},
 	isProductCollectionBlock: true,
 	featured: false,
-	woocommerceOnSale: false,
+	woocommerceOnSale: undefined,
```

Comment:
Should we use `null` instead of `undefined` to indicate the property exists but has no value set?